packages/node-sdk/sm/primitives/evm/rpc/paima-l2.ts

```diff
@@ -185,6 +184,14 @@ export default function* processPaimaL2SyncProtocolResponse(
     PrimitiveEvmRpcPaimaL2Payload,
     response.output.payload,
   );
+  // Fetch the block timestamp (ms) for 24h validation
+  const [blockInfo] = yield* World.resolve(getBlockHeights, {
+    block_heights: [paima_block_height],
+  });
+  const blockTimestampMs = blockInfo?.ms_timestamp
+    ? (blockInfo.ms_timestamp as Date).getTime()
+    : undefined;
```

Comment:
This should just fail if is not available

```suggestion
  const blockTimestampMs = blockInfo!.ms_timestamp.getTime();
```